activerecord/lib/active_record/associations/builder/belongs_to.rb

```diff
@@ -8,7 +8,7 @@ def self.macro
 
     def self.valid_options(options)
       valid = super + [:counter_cache, :optional, :default]
-      valid += [:polymorphic, :foreign_type] if options[:polymorphic]
+      valid += [:polymorphic, :foreign_type] if options.key?(:polymorphic)
```

Comment:
Can you move `:polymorphic` to the above line instead? It is always valid option regardless of the value.

```diff
diff --git a/activerecord/lib/active_record/associations/builder/belongs_to.rb b/activerecord/lib/active_record/associations/builder/belongs_to.rb
index 869b228dc7..584af2c3f2 100644
--- a/activerecord/lib/active_record/associations/builder/belongs_to.rb
+++ b/activerecord/lib/active_record/associations/builder/belongs_to.rb
@@ -7,8 +7,8 @@ def self.macro
     end
 
     def self.valid_options(options)
-      valid = super + [:counter_cache, :optional, :default]
-      valid += [:polymorphic, :foreign_type] if options[:polymorphic]
+      valid = super + [:polymorphic, :counter_cache, :optional, :default]
+      valid += [:foreign_type] if options[:polymorphic]
       valid += [:ensuring_owner_was] if options[:dependent] == :destroy_async
       valid
     end
```

Reply:
done.